dialectRT_test.go

```diff
@@ -152,143 +167,94 @@ func TestDialectRTCommonXML(t *testing.T) {
 	}
 }
 
-func TestDecodeAndEncodeRT0(t *testing.T) {
-	// Encode using CT
-	c := casesMsgsTest[0]
-	dMsgCT, err := newDialectMessage(c.parsed)
-	require.NoError(t, err)
-	bytesEncoded, err := dMsgCT.encode(c.parsed, c.isV2)
-	require.NoError(t, err)
-	require.Equal(t, c.raw, bytesEncoded)
-
-	// Decode bytes using RT
-	includeDirs := []string{"./mavlink-upstream/message_definitions/v1.0"}
-	defs, version, err := libgen.XMLToFields("./mavlink-upstream/message_definitions/v1.0/ardupilotmega.xml", includeDirs)
-	require.NoError(t, err)
-
-	// Create dialect from the parsed defs.
-	dRT, err := NewDialectRT(version, defs)
-	dMsgRT := dRT.messages[332]
-	require.NoError(t, err)
-	require.Equal(t, uint(3), dRT.getVersion())
-
-	// Decode bytes using RT
-	msgDecoded, err := dMsgRT.decode(bytesEncoded, c.isV2)
-	require.NoError(t, err)
-	fmt.Printf("%+v\n", msgDecoded)
+func TestDecodeAndEncodeRT(t *testing.T) {
+	for _, c := range casesMsgsTest {
```

Comment:
You should also add tests for cases that _shouldn't_ work; malformed JSON, objects with the field values inserted into the Fields map that are the wrong type, etc.